src/passes/MergeBlocks.cpp

```diff
@@ -564,7 +564,7 @@ struct MergeBlocks : public WalkerPass<PostWalker<MergeBlocks>> {
 
   void visitCall(Call* curr) { handleCall(curr); }
 
-  void visitCallIndirect(CallIndirect* curr) {
+  template<typename T> void handleNonDirectCall(T* curr) {
```

Comment:
This seems like a code smell that suggests `CallIndirect` and `CallRef` should be rolled into a single expression class (like how `call` and `return_call` are combined), but I'm not sure that would be better overall. Did you consider that approach?